cmake.config.h.in

```diff
@@ -0,0 +1,429 @@
+
+/* Define to one of `_getb67', `GETB67', `getb67' for Cray-2 and Cray-YMP
+   systems. This function is required for `alloca.c' support on those systems.
+   */
+/* #undef CRAY_STACKSEG_END */
+
+/* Define to 1 if using `alloca.c'. */
+/* #undef C_ALLOCA */
+
+/* Define to disable Fortran wrappers. */
+/* #undef DISABLE_FORTRAN */
+
+/* Define to dummy `main' function (if any) required to link to the Fortran
+   libraries. */
+/* #undef F77_DUMMY_MAIN */
+
+/* Define to a macro mangling the given C identifier (in lower and upper
+   case), which must not contain underscores, for linking with Fortran. */
+#define F77_FUNC(name,NAME) name ## _
+
+/* As F77_FUNC, but for C identifiers containing underscores. */
+#define F77_FUNC_(name,NAME) name ## _
+
+/* Define if F77_FUNC and F77_FUNC_ are equivalent. */
+#define F77_FUNC_EQUIV 1
+
+/* Define if F77 and FC dummy `main' functions are identical. */
+/* #undef FC_DUMMY_MAIN_EQ_F77 */
+
+/* C compiler name and flags */
+#define FFTW_CC "gcc -std=gnu99 -O3 -fomit-frame-pointer -mtune=native -malign-double -fstrict-aliasing -fno-schedule-insns -ffast-math"
+
+/* Define to enable extra FFTW debugging code. */
+/* #undef FFTW_DEBUG */
+
+/* Define to enable alignment debugging hacks. */
+/* #undef FFTW_DEBUG_ALIGNMENT */
+
+/* Define to enable debugging malloc. */
+/* #undef FFTW_DEBUG_MALLOC */
+
+/* Define to enable the use of alloca(). */
+#define FFTW_ENABLE_ALLOCA 1
+
+/* Define to compile in long-double precision. */
+#cmakedefine FFTW_LDOUBLE 1
+
+/* Define to compile in quad precision. */
+#cmakedefine FFTW_QUAD 1
+
+/* Define to enable pseudorandom estimate planning for debugging. */
+/* #undef FFTW_RANDOM_ESTIMATOR */
+
+/* Define to compile in single precision. */
+#cmakedefine FFTW_SINGLE 1
+
+/* Define to 1 if you have the `abort' function. */
+#define HAVE_ABORT 1
+
+/* Define to 1 if you have `alloca', as a function or macro. */
+#cmakedefine HAVE_ALLOCA 1
+
+/* Define to 1 if you have <alloca.h> and it should be used (not on Ultrix).
+   */
+#cmakedefine HAVE_ALLOCA_H 1
+
+/* Define to enable Altivec optimizations. */
+/* #undef HAVE_ALTIVEC */
+
+/* Define to 1 if you have the <altivec.h> header file. */
+#cmakedefine HAVE_ALTIVEC_H 1
+
+/* Define if you have enabled the cycle counter on ARMv8 */
+/* #undef HAVE_ARMV8CC */
+
+/* Define to enable AVX optimizations. */
+#cmakedefine HAVE_AVX 1
+
+/* Define to enable AVX2 optimizations. */
+#cmakedefine HAVE_AVX2 1
+
+/* Define to enable AVX512 optimizations. */
+/* #undef HAVE_AVX512 */
+
+/* Define to enable 128-bit FMA AVX optimization */
+/* #undef HAVE_AVX_128_FMA */
+
+/* Define to 1 if you have the `BSDgettimeofday' function. */
+/* #undef HAVE_BSDGETTIMEOFDAY */
+
+/* Define to 1 if you have the `clock_gettime' function. */
+#cmakedefine HAVE_CLOCK_GETTIME
+
+/* Define to 1 if you have the `cosl' function. */
+#define HAVE_COSL 1
+
+/* Define to 1 if you have the <c_asm.h> header file. */
+#cmakedefine HAVE_C_ASM_H 1
```

Comment:
The CMakeLists checks for c_asm.h, then depending on the value of HAVE_C_ASM_H in the CmakeLists the config.h is configured to define or not HAVE_C_ASM_H.